keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/ScalarDoubleTensor.java

```diff
@@ -491,6 +496,23 @@ public DoubleTensor logInPlace() {
         return this;
     }
 
+    /**
+     * This is identical to log().times(y), except that it changes NaN results to 0.
+     * This is important when calculating 0log0, which should return 0
+     * See https://arcsecond.wordpress.com/2009/03/19/0log0-0-for-real/ for some mathematical justification
+     * @param y The tensor value to multiply by
+     * @return the log of this tensor multiplied by y
+     */
+    @Override
+    public DoubleTensor logTimesInPlace(DoubleTensor y) {
+        TensorValidator<Double, Tensor<Double>> validator = TensorValidator.thatChecksForNaN();
```

Comment:
Shouldn't we be checking for 0?

Reply:
No - we allow 0. The logic here is
1. before we do the operation, check there are no `NaN`s in `this` or `y` as if there were it would be wrong to assume it's ok to change them to `0`
2. do the log and times operations
3. change any resulting `NaN` to `0`